### PR TITLE
Adds student number scope to OAuth

### PR DESCRIPTION
**Pull request by ChristopherHammond13**
## What does this PR do?
There has been a request from a student to be able to get a student's ID via OAuth. I've added this as a separate scope. The `oauth/user/data` endpoint will also tell the caller whether the user is a student or not.

Also done some fixes here and there to the OAuth code.

## ✅ Pull Request checklist

- [x] Is this code complete?
- [x] Are docs written for this PR? (if applicable)

## 🚨 Is this a breaking change for API clients?
No

## :squirrel: Deploy notes
Just pull and restart uclapi

### CODE REVIEW

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/1200/
Jenkins statics successfully built.

File: backend/uclapi/oauth/views.py [hunk 1]
```diff
@@ -456,17 +459,32 @@ def token(request):
...
         "given_name": token.user.given_name,
         "cn": token.user.cn,
         "department": token.user.department,
         "upi": token.user.employee_id,
-        "scope_number": token.scope.scope_number
-    }, rate_limiting_data=kwargs)
+        "scope_number": token.scope.scope_number,
+        "is_student": is_student
+    }
+    print("Is student: " + str(is_student))
```
Comment (jermenkoo) [inline backend/uclapi/oauth/views.py]: stray `print`

File: backend/uclapi/oauth/views.py [hunk 2]
```diff
@@ -456,17 +459,32 @@ def token(request):
 @uclapi_protected_endpoint(personal_data=True)
 def userdata(request, *args, **kwargs):
     token = kwargs['token']
+    print("Checking student status")
```
Comment (jermenkoo) [inline backend/uclapi/oauth/views.py]: stray `print`

File: backend/uclapi/oauth/views.py [hunk 3]
```diff
@@ -456,17 +459,32 @@ def token(request):
...
+    try:
+        get_student_by_upi(
+            token.user.employee_id
+        )
+        is_student = True
```
Comment (jermenkoo) [inline backend/uclapi/oauth/views.py]: Not a big fan of this `try: flag = True except: flag = False` pattern but for now it suffices

**Comment by ChristopherHammond13** [comment]
^ironically, a typo in the commit message smh

**Comment by ChristopherHammond13** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.uclapi.com/job/UCLAPI/1201/
Jenkins statics successfully built.

**Pull request merged**